packages/browser-wallet/src/popup/pages/Web3ProofRequest/Display/DisplaySecretStatements.tsx

```diff
@@ -0,0 +1,108 @@
+
+function getStatementValue<Attribute extends AttributeType>(
+    statement: SecretStatementV2,
+    schema: CredentialSchemaSubject,
+    t: TFunction<'web3IdProofRequest', 'displayStatement'>,
+    formatAttribute: (key: string, value: Attribute) => string
+): string {
+    const name = getPropertyTitle(statement.attributeTag, schema);
+    if (statement.type === StatementTypes.AttributeInRange) {
+        return t('proofs.range', {
+            name,
+            upper: formatAttribute(statement.attributeTag, statement.upper as Attribute),
+            lower: formatAttribute(statement.attributeTag, statement.lower as Attribute),
+        });
+    }
+    if (statement.type === StatementTypes.AttributeInSet) {
+        return t('proofs.membership', { name });
+    }
+    if (statement.type === StatementTypes.AttributeNotInSet) {
+        return t('proofs.nonMembership', { name });
+    }
+
+    throw new Error('Unknown statement type');
+}
+
+function getStatementDescription<Attribute extends AttributeType>(
+    statement: SecretStatementV2,
+    schema: CredentialSchemaSubject,
+    t: TFunction<'web3IdProofRequest', 'displayStatement'>,
+    formatAttribute: (key: string, value: Attribute) => string
+) {
+    const name = getPropertyTitle(statement.attributeTag, schema);
+    const listToString = (list: AttributeType[]) =>
+        list.map((member) => formatAttribute(statement.attributeTag, member as Attribute)).join(', ');
+
+    switch (statement.type) {
+        case StatementTypes.AttributeInRange:
+            return t('descriptions.range', {
+                name,
+                upper: formatAttribute(statement.attributeTag, statement.upper as Attribute),
+                lower: formatAttribute(statement.attributeTag, statement.lower as Attribute),
+            });
+        case StatementTypes.AttributeInSet:
+            return t('descriptions.membership', { name, setNames: listToString(statement.set) });
+        case StatementTypes.AttributeNotInSet:
+            return t('descriptions.nonMembership', { name, setNames: listToString(statement.set) });
+        default:
+            throw new Error(`Unknown statement type encountered: ${statement.type}`);
+    }
+}
+
+export function DisplaySecretStatements<Attribute extends AttributeType>({
+    schema,
+    statements,
+    className,
+    formatAttribute = (_, value) => value.toString(),
+}: DisplayProps<SecretStatementV2, Attribute>) {
+    const { t } = useTranslation('web3IdProofRequest', { keyPrefix: 'displayStatement' });
+    const header = t('headers.secret');
+
+    const lines = statements.map((s) => {
+        const value = getStatementValue(s, schema, t, formatAttribute);
+        const title = getPropertyTitle(s.attributeTag, schema);
+        const description = getStatementDescription(s, schema, t, formatAttribute);
+        return {
+            attribute: title,
+            value: value.toString() ?? 'Unavailable',
```

Comment:
Should the value be formatted here with `formatAttribute`?

Reply:
No, value is always/already a string, changed it.